plugin/evm/vm.go

```diff
@@ -379,9 +379,15 @@ func (vm *VM) Initialize(
 		vm.chainConfig.UpgradeConfig = upgradeConfig
 	}
 
-	// create genesisHash after applying upgradeBytes in case
+	// create genesisBlock after applying upgradeBytes in case
 	// upgradeBytes modifies genesis.
-	vm.genesisHash = vm.ethConfig.Genesis.ToBlock(nil).Hash()
+	genesisBlock := vm.ethConfig.Genesis.ToBlock(nil)
+	vm.genesisHash = genesisBlock.Hash()
+
+	// check that subnetEVM upgrade is enabled from genesis
+	if !vm.chainConfig.IsSubnetEVM(genesisBlock.Timestamp()) {
+		return fmt.Errorf("SubnetEVM upgrade is not enabled in genesis")
```

Comment:
Can we require this is set in the genesisBytes and not allow the option to set it via the upgradeBytes to simplify?

Reply:
yes